util/mock/context.go

```diff
@@ -49,12 +51,30 @@ func (c *Context) ClearValue(key fmt.Stringer) {
 
 // GetTxn implements context.Context GetTxn interface.
 func (c *Context) GetTxn(forceNew bool) (kv.Transaction, error) {
-	return nil, nil
+	if c.Store == nil {
+		return nil, nil
+	}
+
+	if c.txn != nil {
+		return c.txn, nil
+	}
+
+	var err error
+	c.txn, err = c.Store.Begin()
+
+	return c.txn, err
 }
 
 // FinishTxn implements context.Context FinishTxn interface.
 func (c *Context) FinishTxn(rollback bool) error {
-	return nil
+	if c.Store == nil || c.txn == nil {
```

Comment:
no need to check c.Store == nil here, if store is nil, the txn must always be nil.